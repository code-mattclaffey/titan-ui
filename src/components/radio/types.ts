import { GenericComponent } from '../../module';

export interface RadioInterface extends GenericComponent {
    id: string;
    name: string;
    required?: boolean;
    labelText: string;
    checked?: boolean;
    disabled?: boolean;
}
