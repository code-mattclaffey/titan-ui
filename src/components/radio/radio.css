/* Checkbox Styling */

:root {
    --radio-indicator-size: 1em;
    --radio-indicator-border-radius: 50%;
    --radio-indicator-margin-right: 0.5em;
    --radio-indicator-top-position: 0.2em;
    --radio-indicator-border: var(--input-border);
    --radio-indicator-selected-size: 0.5em;
    --radio-indicator-selected-border: var(--input-border-focus);
    --radio-indicator-disabled-bg: var(--keyline);
    --radio-indicator-disabled-border-color: var(--keyline);
}

.e-radio {
    cursor: pointer;
    display: flex;
    flex-direction: row-reverse;
    justify-content: flex-end;
}

.e-radio__indicator {
    border: var(--input-border);
    border-radius: var(--radio-indicator-border-radius);
    flex: none;
    height: var(--radio-indicator-size);
    margin-right: var(--radio-indicator-margin-right);
    position: relative;
    top: var(--radio-indicator-top-position);
    width: var(--radio-indicator-size);
}

.e-radio__input:checked ~ .e-radio__indicator {
    border-color: var(--radio-indicator-selected-border);
}

.e-radio__input:disabled:checked ~ .e-radio__indicator,
.e-radio__input:disabled ~ .e-radio__indicator {
    background: var(--radio-indicator-disabled-bg);
    border-color: var(--radio-indicator-disabled-border-color);
}

.e-radio__input:checked ~ .e-radio__indicator::after {
    background: var(--primary);
    border-radius: var(--radio-indicator-border-radius);
    content: '';
    height: var(--radio-indicator-selected-size);
    left: 50%;
    position: absolute;
    top: 50%;
    transform: translate(-50%, -50%);
    width: var(--radio-indicator-selected-size);
}

.e-radio__input:disabled:checked ~ .e-radio__indicator::after,
.e-radio__input:disabled ~ .e-radio__indicator::after {
    display: none;
}
