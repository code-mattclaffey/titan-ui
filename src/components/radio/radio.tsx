import React from 'react';
import classNames from 'classnames';
import { RadioInterface } from './types';
import { Label } from '../label/label';

export const Radio = ({
    additionalClassNames = '',
    name,
    id,
    checked = false,
    required = true,
    labelText,
    disabled,
}: RadioInterface) => {
    const classes = classNames('e-radio', additionalClassNames);

    return (
        <Label
            htmlFor={id}
            additionalClassNames={classes}
            labelText={labelText}
            required={false}
        >
            <input
                type="radio"
                className="u-visually-hidden e-radio__input"
                name={name}
                id={id}
                required={required}
                defaultChecked={checked}
                disabled={disabled}
            />
            <span className="e-radio__indicator" />
        </Label>
    );
};
