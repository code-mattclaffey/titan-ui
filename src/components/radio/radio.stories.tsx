import React from 'react';

import { Radio } from './radio';

export default {
    title: 'Radio',
    componentSubtitle: '',
};

export const radio = () => (
    <>
        <Radio
            name="hello-test"
            id="hello-test-radio-1"
            labelText="Radio One"
        />
        <Radio
            name="hello-test"
            id="hello-test-radio-box-2"
            labelText="Radio Two"
            checked
        />
        <Radio
            name="hello-test"
            id="hello-test-radio-box-3"
            labelText="Radio Three (Disabled)"
            disabled
        />
    </>
);
