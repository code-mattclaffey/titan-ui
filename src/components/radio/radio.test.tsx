import React from 'react';
import { render, cleanup } from '@testing-library/react';
import { Radio } from './radio';

afterEach(() => {
    cleanup();
});

describe('<Radio />', () => {
    it('should match snapshot in default state', () => {
        const { container } = render(
            <Radio
                name="hello-test"
                id="hello-test"
                labelText="Hello Test"
            />,
        );

        expect(container).toMatchSnapshot();
    });

    it('should match snapshot in checked state', () => {
        const { container } = render(
            <Radio
                name="hello-test"
                id="hello-test"
                labelText="Hello Test"
                checked
            />,
        );

        expect(container).toMatchSnapshot();
    });

    it('should match snapshot in disabled state', () => {
        const { container } = render(
            <Radio
                name="hello-test"
                id="hello-test"
                labelText="Hello Test"
                disabled
            />,
        );

        expect(container).toMatchSnapshot();
    });
});
