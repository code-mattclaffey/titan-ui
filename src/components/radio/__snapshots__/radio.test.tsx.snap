// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Radio /> should match snapshot in checked state 1`] = `
<div>
  <label
    class="e-label e-radio"
    data-testid="e-label"
    for="hello-test"
  >
    <span
      class="e-label__text"
    >
      Hello Test
    </span>
    <input
      checked=""
      class="u-visually-hidden e-radio__input"
      id="hello-test"
      name="hello-test"
      required=""
      type="radio"
    />
    <span
      class="e-radio__indicator"
    />
  </label>
</div>
`;

exports[`<Radio /> should match snapshot in default state 1`] = `
<div>
  <label
    class="e-label e-radio"
    data-testid="e-label"
    for="hello-test"
  >
    <span
      class="e-label__text"
    >
      Hello Test
    </span>
    <input
      class="u-visually-hidden e-radio__input"
      id="hello-test"
      name="hello-test"
      required=""
      type="radio"
    />
    <span
      class="e-radio__indicator"
    />
  </label>
</div>
`;

exports[`<Radio /> should match snapshot in disabled state 1`] = `
<div>
  <label
    class="e-label e-radio"
    data-testid="e-label"
    for="hello-test"
  >
    <span
      class="e-label__text"
    >
      Hello Test
    </span>
    <input
      class="u-visually-hidden e-radio__input"
      disabled=""
      id="hello-test"
      name="hello-test"
      required=""
      type="radio"
    />
    <span
      class="e-radio__indicator"
    />
  </label>
</div>
`;
